dotCMS/src/integration-test/java/com/dotcms/security/secret/ServiceIntegrationAPIImplTest.java

```diff
@@ -0,0 +1,302 @@
+        final ServiceIntegrationAPI api = APILocator.getServiceIntegrationAPI();
+        api.saveSecrets(bean, host, admin);
+        final Optional<ServiceSecrets> optionalBean = api
+                .getSecretsForService(serviceKey, host, admin);
+
+        assertTrue(optionalBean.isPresent());
+
+        final ServiceSecrets recoveredBean = optionalBean.get();
+        assertEquals(serviceKey, recoveredBean.getServiceKey());
+        assertTrue(recoveredBean.getSecrets().containsKey("mySecret1"));
+        assertTrue(recoveredBean.getSecrets().containsKey("mySecret2"));
+        assertTrue(recoveredBean.getSecrets().containsKey("boolSecret1"));
+        assertTrue(recoveredBean.getSecrets().containsKey("boolSecret2"));
+
+        final Secret secret1 = recoveredBean.getSecrets().get("mySecret1");
+        assertEquals("Once I saw a UFO", secret1.getString());
+        assertTrue(secret1.isHidden());
+
+        final Secret secret2 = recoveredBean.getSecrets().get("mySecret2");
+        assertEquals("Football soccer sucks!", secret2.getString());
+        assertTrue(secret2.isHidden());
+
+        final Secret boolSecret1 = recoveredBean.getSecrets().get("boolSecret1");
+        assertTrue(boolSecret1.getBoolean());
+        assertFalse(boolSecret1.isHidden());
+
+        final Secret boolSecret2 = recoveredBean.getSecrets().get("boolSecret2");
+        assertFalse(boolSecret2.getBoolean());
+        assertFalse(boolSecret2.isHidden());
+    }
+
+    @Test
+    public void Test_Store_Json_On_Different_Sites_Then_Pull_Sites_By_Host_Verify_Match()
+            throws DotDataException, DotSecurityException {
+
+        final User admin = TestUserUtils.getAdminUser();
+        final Host host1 = new SiteDataGen().nextPersisted();
+        final Host host2 = new SiteDataGen().nextPersisted();
+
+        final ServiceSecrets.Builder builder1 = new ServiceSecrets.Builder();
+
+        final ServiceSecrets bean1Host1 = builder1.withServiceKey("serviceKey-1-Host-1")
+                .withHiddenSecret("h1:secret1", "sec1")
+                .withHiddenSecret("h1:secret2", "sec2")
+                .withSecret("h1:bool1", true)
+                .withSecret("h1:bool2", false)
+                .build();
+
+        final ServiceSecrets bean2Host1 = builder1.withServiceKey("serviceKey-2-Host-1")
+                .withHiddenSecret("h1:secret1", "sec1")
+                .withHiddenSecret("h1:secret2", "sec2")
+                .withSecret("h1:bool1", true)
+                .withSecret("h1:bool2", false)
+                .build();
+
+        final ServiceSecrets bean3Host1 = builder1.withServiceKey("serviceKey-3-Host-1")
+                .withHiddenSecret("h1:secret1", "sec1")
+                .withHiddenSecret("h1:secret2", "sec2")
+                .withSecret("h1:bool1", true)
+                .withSecret("h1:bool2", false)
+                .build();
+
+        ServiceSecrets.Builder builder2 = new ServiceSecrets.Builder();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'builder2' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)